File: Uva/902.java
import java.math.*;
import java.util.*;
import java.math.*;
import java.util.regex.*;

class Main {
	public static void main(String[] args) {
		HashMap<String,Integer> mmap = new HashMap<String,Integer>();
		Scanner in = new Scanner(System.in);
		while (in.hasNext()) {
			int total = 0;
			String result = "";
			int n = in.nextInt();
			String line = in.next();
			mmap.clear();
			for (int i=0;i<=line.length()-n;i++) {
				int count;
				String token = line.substring(i,i+n);
				if (mmap.containsKey(token)) {
					int val = mmap.get(token);
					mmap.put(token,val+1);
					count = val+1;
				}
				else {
					mmap.put(token,1);
					count = 1;
				}
				if (count>total) {
					total = count;
					result = token;
				}
			}
			System.out.println(result);
		}
	}
}

File: Uva/902.cpp
#include <iostream>
#include <cstdio>
#include <algorithm>
#include <cstring>
#include <string>
#include <cmath>
#include <map>

using namespace std;
map<string,int> mmap;
int main() {
	int n;
	string line;
	while (scanf("%d",&n)!=EOF) {
		int count = 0;
		string temp = "";
		cin >> line;
		mmap.clear();
		for (int i=0;i<=line.size()-n;i++) {
			string token = line.substr(i,n);
			mmap[token]++;
			if (mmap[token]>count) {
				count = mmap[token];
				temp = token;
			}
		}
		cout << temp << endl;
		
	}
	return 0;
}